iamctl/pkg/identityProviders/export.go

```diff
@@ -0,0 +1,139 @@
+/**
+* Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com) All Rights Reserved.
+*
+* WSO2 LLC. licenses this file to you under the Apache License,
+* Version 2.0 (the "License"); you may not use this file except
+* in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied. See the License for the
+* specific language governing permissions and limitations
+* under the License.
+ */
+
+package identityproviders
+
+import (
+	"crypto/tls"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"mime"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/wso2-extensions/identity-tools-cli/iamctl/pkg/utils"
+)
+
+func ExportAll(exportFilePath string, format string) {
+
+	// Export all identity providers to the IdentityProviders folder.
+	log.Println("Exporting identity providers...")
+	exportFilePath = filepath.Join(exportFilePath, utils.IDENTITY_PROVIDERS)
+	os.MkdirAll(exportFilePath, 0700)
+
+	idps, err := getIdpList()
+	if err != nil {
+		log.Println("Error: when exporting identity providers.", err)
+	} else {
+		excludeSecerts := utils.AreSecretsExcluded(utils.TOOL_CONFIGS.IdpConfigs)
+		for _, idp := range idps {
+			if !utils.IsResourceExcluded(idp.Name, utils.TOOL_CONFIGS.IdpConfigs) {
+				log.Println("Exporting identity provider: ", idp.Name)
+
+				err := exportIdp(idp.Id, exportFilePath, format, excludeSecerts)
+				if err != nil {
+					log.Printf("Error while exporting identity providers: %s. %s", idp.Name, err)
+				} else {
+					log.Println("Identity provider exported successfully: ", idp.Name)
+				}
+			}
+		}
+	}
+}
+
+func exportIdp(idpId string, outputDirPath string, format string, excludeSecrets bool) error {
+
+	var fileType string
+	// TODO: Extend support for json and xml formats.
+	switch format {
+	case "json":
+		fileType = utils.MEDIA_TYPE_JSON
+	case "xml":
+		fileType = utils.MEDIA_TYPE_XML
+	default:
+		fileType = utils.MEDIA_TYPE_YAML
+	}
+
+	var reqUrl = utils.SERVER_CONFIGS.ServerUrl + "/t/" + utils.SERVER_CONFIGS.TenantDomain + "/api/server/v1/identity-providers/file/" + idpId
+
+	var err error
+	req, err := http.NewRequest("GET", reqUrl, strings.NewReader(""))
+	if err != nil {
+		return fmt.Errorf("error while creating the request to export identity provider: %s", err)
+	}
+	req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
```

Comment:
Make this content type also a constant. Go might already have provided these content types in their default http library check on that also.

Reply:
http library does not have constants for form media type. Therefore added a new constant.